pkg/plugin/service.go

```diff
@@ -70,3 +70,45 @@ func (r *Service) List(ctx context.Context) (map[string]Specification, error) {
 
 	return specs, nil
 }
+
+func (r *Service) ValidateSourceConfig(ctx context.Context, d Dispenser, settings map[string]string) error {
+	src, err := d.DispenseSource()
+	if err != nil {
+		return cerrors.Errorf("could not dispense source: %w", err)
+	}
+
+	defer func(src SourcePlugin, ctx context.Context) {
+		err := src.Teardown(ctx)
+		if err != nil {
+			return
+		}
+	}(src, ctx)
```

Comment:
We lose the error returned by `Teardown`. A way to solve it would be to create a named return and overwrite the returned error if we aren't returning an error already ([example](https://go.dev/play/p/LIg9tLSjPnB)).

Nitpick: we don't need to specify params for this func, it can access `src` and `ctx` already.
```suggestion
	defer func() {
		terr := src.Teardown(ctx)
		if err == nil {
			err = terr // only overwrite if we are not returning an error
		}
	}()
```